packages/firebase_auth/firebase_auth/ios/Classes/FLTFirebaseAuthPlugin.m

```diff
@@ -839,13 +839,23 @@ - (void)userUpdateProfile:(id)arguments withMethodCallResult:(FLTFirebaseMethodC
   NSDictionary *profileUpdates = arguments[@"profile"];
   FIRUserProfileChangeRequest *changeRequest = [currentUser profileChangeRequest];
 
-  if (profileUpdates[@"displayName"] != nil &&
-      ![profileUpdates[@"displayName"] isEqual:[NSNull null]]) {
-    changeRequest.displayName = profileUpdates[@"displayName"];
+  if (profileUpdates[@"displayName"] != nil) {
+    if ([profileUpdates[@"displayName"] isEqual:[NSNull null]]) {
+      changeRequest.displayName = nil;
+    } else {  
+      changeRequest.displayName = profileUpdates[@"displayName"];
+    }
   }
 
-  if (profileUpdates[@"photoURL"] != nil && ![profileUpdates[@"photoURL"] isEqual:[NSNull null]]) {
-    changeRequest.photoURL = [NSURL URLWithString:profileUpdates[@"photoURL"]];
+  if (profileUpdates[@"photoURL"] != nil) {
+    if ([profileUpdates[@"photoURL"] isEqual:[NSNull null]]) {
+      // We apparently cannot set photoURL to nil/NULL to remove it.
```

Comment:
There are undocumented methods for setting these properties:
```obj-c
[changeRequest setDisplayName:nil];
[changeRequest setPhotoURL:nil];
```
that should work. Reference [here](https://github.com/firebase/firebase-ios-sdk/blob/master/FirebaseAuth/Sources/User/FIRUser.m#L1521-L1548)

Reply:
It doesn't seem to work either

Reply:
fair enough, I'll resolve this then 👍